backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -798,15 +807,43 @@ export class ExperimentService {
           partition.experiment = experimentDoc;
           return partition;
         });
-
+      
+      // creating queries docs
+      const promiseArray = [];
+      let queryDocsToSave =
+        (queries &&
+          queries.length > 0 &&
+          queries.map((query: Query) => {
+            promiseArray.push(this.metricRepository.findOne(query.metric));
+            // tslint:disable-next-line:no-shadowed-variable
+            const { createdAt, updatedAt, versionNumber, metric, ...rest } = query;
+            rest.experiment = experimentDoc;
+            rest.id = rest.id || uuid();
+            return rest;
+          })) ||
+        [];
+      if (promiseArray.length) {
+        const metricsDocs = await Promise.all([...promiseArray]);
```

Comment:
`[...promiseArray]` why you creating new array by spreading existing array?

Reply:
We are taking it in new metricDocs array to use ahead. You have also used the same in the same `ExperimentService.ts` file in `updateExperimentInDB()` function as shown below:

<img width="807" alt="Screenshot 2021-11-29 at 12 30 37 AM" src="https://user-images.githubusercontent.com/33730817/143782135-5f2e9d83-90ba-4cb4-b683-39949b9d1914.png">

Let me know if there is something wrong, I would change at both places.

Reply:
Yup that is incorrect correct at both the places